File: org.kevoree.modeling.microframework.typescript/src/main/ts/org/kevoree/modeling/api/select/KQueryParam.ts

class KQueryParam {

  private _name: string = null;
  private _value: string = null;
  private _negative: boolean = null;

  constructor(p_name: string, p_value: string, p_negative: boolean) {
    this._name = p_name;
    this._value = p_value;
    this._negative = p_negative;
  }

  public name(): string {
    return this._name;
  }

  public value(): string {
    return this._value;
  }

  public isNegative(): boolean {
    return this._negative;
  }

}

File: org.kevoree.modeling.microframework/src/main/java/org/kevoree/modeling/api/traversal/selector/KQueryParam.java
package org.kevoree.modeling.api.traversal.selector;

/**
 * Created by duke on 10/24/14.
 */
public class KQueryParam {

    private String _name;
    private String _value;
    private boolean _negative;

    public KQueryParam(String p_name, String p_value, boolean p_negative) {
        this._name = p_name;
        this._value = p_value;
        this._negative = p_negative;
    }

    public String name() {
        return _name;
    }

    public String value() {
        return _value;
    }

    public boolean isNegative() {
        return _negative;
    }

}
